metadata/src/main/java/org/apache/kafka/controller/ConfigurationControlManager.java

```diff
@@ -269,12 +269,12 @@ private ApiError validateAlterConfig(ConfigResource configResource,
                                          List<ApiMessageAndVersion> recordsImplicitlyDeleted,
                                          boolean newlyCreatedResource) {
         Map<String, String> allConfigs = new HashMap<>();
-        Map<String, String> existingConfigsInMap = new HashMap<>();
+        Map<String, String> existingConfigsMap = new HashMap<>();
         Map<String, String> alteredConfigsForAlterConfigPolicyCheck = new HashMap<>();
-        TimelineHashMap<String, String> existingConfigs = configData.get(configResource);
-        if (existingConfigs != null) {
-            existingConfigsInMap.putAll(existingConfigs);
-            allConfigs.putAll(existingConfigs);
+        TimelineHashMap<String, String> existingConfigsSnapshot = configData.get(configResource);
```

Comment:
for my understanding, why we are not using the `existingConfigsSnapshot` directly?